### DISCUSSION TITLE
Show gaps only when the modifier key is pressed

### DISCUSSION THREAD

**Discussion by TheMcnafaha** [i3/i3#6134] [Q&A] [answered]
Hey fellow i3 users! 

As the tittle says, I'm looking to see if its possible to show gaps only when the modifier key is pressed and then reset the gaps once the key is let go.

My ideal is that I can have gaps on when  I want to change windows. That's the one time were  I really want gaps, so if I could make the modifier key turn them on while maintaining all of its default behavior that would make my workflow super smooth.  

I've read both gaps documentation and  the provided config. So I know that what I want is quite simple to do with two keys. I just don't know how to do it with just the modifier key.

 The best solution that I could come up with was running a script that calls the i3 gaps config anytime I press the modifier key and then makes one last call that resets the gaps config when the key is no longer being pressed. I'm not a big fan of this approach as it essentially splits my config in two. So to restate the tittle, is there a way to accomplish this with just using the i3 config file?

**Comment by k-laig** [comment]


**Reply by k-laig** [reply]
There isn't its just some scamy antivirus that bought and redirected it for marketing. Plz someone host this, it would be fun.

**Reply by TheMcnafaha** [reply]
Thanks, I'll try to implement it and post the solution if it works!

**Reply by orestisfl** [reply]
Please don't post ChatGPT answers you cannot verify, this can lead to big time waste for readers and maintainers. The answer is completely wrong as is the concept. You can't re-define or change i3 variables and you can't call exec inside `mode`s. The generated configuration has syntax errors.

I am marking this answer as spam.

**Comment by orestisfl** [comment, marked as answer]
Something like that?

```
bindsym Super_L gaps inner all set 20
bindsym --release Super_L gaps inner all set 0
```

**Reply by orestisfl** [reply]
A limitation of this is that executing a different mod-based binding does not reset the gaps to 0

**Reply by TheMcnafaha** [reply]
Works perfectly! Thank you so much, that prefix seem really useful.
